packages/headless/src/lib/headless.service.ts

```diff
@@ -0,0 +1,531 @@
+  QueryObserverOptions,
+  QueryObserverResult,
+  MutationObserverResult,
+} from '@tanstack/query-core';
+import io from 'socket.io-client';
+import { ApiService, IUserPreferenceSettings, IStoreQuery } from '@novu/client';
+import { IOrganizationEntity, IMessage } from '@novu/shared';
+
+import { QueryService } from './query.service';
+import type { ISession } from '../utils/types';
+import {
+  NOTIFICATIONS_QUERY_KEY,
+  ORGANIZATION_QUERY_KEY,
+  SESSION_QUERY_KEY,
+  UNSEEN_COUNT_QUERY_KEY,
+  USER_PREFERENCES_QUERY_KEY,
+} from '../utils';
+import {
+  FetchResult,
+  IHeadlessServiceOptions,
+  IUpdateActionVariables,
+  IUpdateUserPreferencesVariables,
+  UpdateResult,
+} from './types';
+
+export const NOTIFICATION_CENTER_TOKEN_KEY = 'nc_token';
+const isBrowser = typeof window !== 'undefined';
+const SESSION_NOT_INITIALIZED_ERROR =
+  'Session is not initialized, please use the initializeSession method first';
+
+const getToken = (): string | null => {
+  if (isBrowser) {
+    return localStorage.getItem(NOTIFICATION_CENTER_TOKEN_KEY);
+  }
+
+  return null;
+};
+
+export class HeadlessService {
+  private api: ApiService;
+  private queryClient: QueryClient = null;
+  private queryService: QueryService;
+  private socket: {
+    on: (event: string, listener: (data?: unknown) => void) => void;
+    off: (event: string) => void;
+    disconnect: () => void;
+  } | null = null;
+  private session: ISession | null = null;
+  private sessionQueryOptions: QueryObserverOptions<ISession, unknown> = {
+    queryKey: SESSION_QUERY_KEY,
+    cacheTime: Infinity,
+    staleTime: Infinity,
+    queryFn: () =>
+      this.api.initializeSession(
+        this.options.applicationIdentifier,
+        this.options.subscriberId,
+        this.options.subscriberHash
+      ),
+  };
+  private organizationQueryOptions: QueryObserverOptions<
+    IOrganizationEntity,
+    unknown
+  > = {
+    queryKey: ORGANIZATION_QUERY_KEY,
+    cacheTime: Infinity,
+    staleTime: Infinity,
+    queryFn: () => this.api.getOrganization(),
+  };
+  private unseenCountQueryOptions: QueryObserverOptions<
+    { count: number },
+    unknown
+  > = {
+    queryKey: UNSEEN_COUNT_QUERY_KEY,
+    queryFn: () => this.api.getUnseenCount(),
+  };
+  private userPreferencesQueryOptions: QueryObserverOptions<
+    IUserPreferenceSettings[],
+    unknown
+  > = {
+    queryKey: USER_PREFERENCES_QUERY_KEY,
+    queryFn: () => this.api.getUserPreference(),
+  };
+
+  constructor(private options: IHeadlessServiceOptions) {
+    const backendUrl = options.backendUrl ?? 'https://api.novu.co';
+    const token = getToken();
+    this.api = new ApiService(backendUrl);
+    this.applyToken(token);
+
+    this.queryClient = new QueryClient({
+      defaultOptions: {
+        queries: {
+          refetchOnWindowFocus: false,
+          retry: options?.config?.retry ?? 3,
+          ...(options?.config?.retryDelay
+            ? { retryDelay: options?.config?.retryDelay }
+            : {}),
```

Comment:
```suggestion
          ...(options?.config?.retryDelay && { retryDelay: options?.config?.retryDelay }),
```
You can simplify and not need a ternary.
```
const a = {
   ...(false && { obj }),
}
```
Does not throw an error and would be the same as not spreading anything or spreading an empty object.

Reply:
thanks! will change it :)